utils/build-script

```diff
@@ -137,15 +137,10 @@ class HostSpecificConfiguration(object):
             # For platforms which normally require a connected device to
             # test, the default behavior is to run tests that only require
             # the host (i.e., they do not attempt to execute).
-            if deployment_platform.is_darwin and \
-               deployment_platform.is_embedded and \
-               not deployment_platform.is_simulator:
-                if deployment_platform not in \
+            if deployment_platform.uses_host_tests and \
+                    deployment_platform not in \
                         invocation.platforms_to_skip_test_host:
-                    test_host_only = True
-                    test = True
-                else:
-                    test = False
+                test_host_only = True
```

Comment:
This is another change of behaviour. Since I don’t know how the iOS/tvOS/watchOS tests are actually run, I hope this change doesn’t break anything. The most important part is removing the `test = False` from the `else:` branch. That branch made impossible to test the device tests in my experience. I simply let the `test` variable to take the value that was assigned above.